src/uproot/interpretation/strings.py

```diff
@@ -152,74 +157,150 @@ def basket_array(
             cursor_offset=cursor_offset,
             library=library,
         )
+        if (
+            isinstance(library, uproot.interpretation.library.Awkward)
+            and byte_offsets is None
+        ):
+            import awkward.forth
 
-        if byte_offsets is None:
-            counts = numpy.empty(len(data), dtype=numpy.int32)
-            outdata = numpy.empty(len(data), dtype=data.dtype)
-
-            pos = 0
-            entry_num = 0
-            len_outdata = 0
-
+            self._special_case = True
             if self._length_bytes == "1-5":
-                while True:
-                    if pos >= len(data):
-                        break
-                    size = data[pos]
-                    pos += 1
-                    if size == 255:
-                        (size,) = _string_4byte_size.unpack(data[pos : pos + 4])
-                        pos += 4
-                    counts[entry_num] = size
-                    entry_num += 1
-                    outdata[len_outdata : len_outdata + size] = data[pos : pos + size]
-                    len_outdata += size
-                    pos += size
+                if not self._code_complete:
+                    self._forth_code = """
+                        input stream
+
+                        output out-main uint8
+                        output out-offsets int64
+
+                        0 out-offsets <- stack
+
+                        begin
+                            stream !B-> stack dup 255 = if drop stream !I-> stack then dup out-offsets +<- stack stream #!B-> out-main
+                        again"""
+
+                    self._code_complete = True
```

Comment:
You don't have to be careful about not regenerating the "1-5" Forth code string and the "4" Forth code string, since they're constants (`.pyc`-compile-time constants for Python: they're not made, they're just looked up).